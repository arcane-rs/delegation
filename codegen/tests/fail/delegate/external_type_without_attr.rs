use delegation::{private::Either, delegate};

#[delegate(for(for<T: AsStr> Either<T, T>))]
trait AsStr {
    fn as_str(&self) -> &str;
}

impl AsStr for String {
    fn as_str(&self) -> &str {
        self.as_str()
    }
}

fn main() {
    unreachable!()
}
