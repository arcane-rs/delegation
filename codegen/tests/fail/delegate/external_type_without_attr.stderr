error[E0277]: the trait bound `Either<T, T>: Convert` is not satisfied
 --> tests/fail/delegate/external_type_without_attr.rs:3:1
  |
3 | #[delegate(for(for<T: AsStr> Either<T, T>))]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Convert` is not implemented for `Either<T, T>`
  |
  = help: the trait `AsStr` is implemented for `delegation::private::Wrapper<__Delegate>`
note: required for `delegation::private::Wrapper<Either<T, T>>` to implement `AsStr`
 --> tests/fail/delegate/external_type_without_attr.rs:3:1
  |
3 | #[delegate(for(for<T: AsStr> Either<T, T>))]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
4 | trait AsStr {
  |       ^^^^^
  = note: this error originates in the attribute macro `delegate` (in Nightly builds, run with -Z macro-backtrace for more info)
